hoot-services/src/main/java/hoot/services/controllers/grail/GrailResource.java

```diff
@@ -330,30 +332,40 @@ public Response differentialStats(@Context HttpServletRequest request,
             File tagDiffFile = new File(workDir, "diff.tags.osc");
             jobInfo.put("hasTags", tagDiffFile.exists());
         }
-        catch (TransformerException e) {
-            throw new RuntimeException("Error invoking XPathAPI!", e);
+        catch (IllegalArgumentException e) {
+            throw new WebApplicationException(e, Response.serverError().entity("Changeset file not found.").build());
         }
-        catch (Exception exc) {
-            String msg = "Error during differential stats! jobid: " + jobDir;
-            throw new WebApplicationException(exc, Response.serverError().entity(msg).build());
+        catch (IOException e) {
+            throw new WebApplicationException(e, Response.serverError().entity("Changeset file not found.").build());
+        }
+        catch (ParserConfigurationException e) {
+            throw new WebApplicationException(e, Response.serverError().entity("Changeset file is malformed.").build());
+        }
+        catch (SAXException e) {
+            throw new WebApplicationException(e, Response.serverError().entity("Changeset file is malformed.").build());
+        }
+        catch (TransformerException e) {
+            throw new WebApplicationException(e, Response.serverError().entity("Changeset file is malformed.").build());
         }
 
+
         return Response.ok(jobInfo.toJSONString()).build();
     }
 
     /**
-     * Pushes the specified differential changeset back to the
+     * Pushes the specified changeset back to the
      * reference dataset API
      *
      * Takes in a json object
-     * POST hoot-services/grail/differentialpush
+     * POST hoot-services/grail/changesetpush
```

Comment:
renamed a bunch of "differential" stuff to "changeset" to reflect the methods are not just for the "differential changeset" workflow